src/UI/Art/SpriteArtist/PlayerSetupCommanderArtist.java

```diff
@@ -0,0 +1,178 @@
+    Graphics myG = image.getGraphics();
+
+    // Allocate space for the Commander faces.
+    int cursorSizePx = SpriteLibrary.getCursorSprites().getFrame(0).getWidth(); // Cursor frames are square.
+    int coZoneYCenter = myHeight / 2;
+
+    /////////////// Tooltip ////////////////////////////
+    BufferedImage tooltip = SpriteUIUtils.makeTextFrame("Press Q for more info", 3, 2);
+    myG.drawImage(tooltip, myWidth - tooltip.getWidth(), 3, null);
+
+    /////////////// Commander Portrait //////////////////////
+    int highlightedCommander = control.getSelectedCommander();
+    BufferedImage likeness = SpriteLibrary.getCommanderSprites( infos.get(highlightedCommander).name ).body;
+    myG.drawImage(likeness, myWidth-likeness.getWidth(), myHeight-likeness.getHeight(), null);
+
+    /////////////// Team Panels //////////////////////
+    // Calculate the vertical space each player panel will consume.
+    int panelBuffer = 3;
+    int panelHeight = CommanderPanel.PANEL_HEIGHT+panelBuffer;
+
+    // If we are moving from one option to another, calculate the intermediate draw location.
+//    if( animHighlightedPlayer != highlightedPlayer )
+//    {
+//      double slide = SpriteUIUtils.calculateSlideAmount(animHighlightedPlayer, highlightedPlayer);
+//      animHighlightedPlayer += slide;
+//    }
+
+    // Find where the zeroth Commander should be drawn. Start by assuming it's centered.
+    int drawYCenter = coZoneYCenter;
+
+    // We're gonna make this an endless scroll, so back up (in y-space and in the CO list) until
+    // we find the Commander that would be off the top of the screen.
+    OptionSelector coToDraw = new OptionSelector(infos.size());
+    coToDraw.setSelectedOption(highlightedCommander);
+    while( drawYCenter + CommanderPanel.PANEL_HEIGHT/2 > 0 )
+    {
+      coToDraw.handleInput(InputAction.UP);
+      drawYCenter -= panelHeight;
+    }
+    // We don't actually want to draw something off the screen, so go forward one again.
+    coToDraw.handleInput(InputAction.DOWN);
+    drawYCenter += panelHeight;
+
+    // Draw all of the visible commander panels.
+    for(; drawYCenter - CommanderPanel.PANEL_HEIGHT/2 < myHeight ; coToDraw.handleInput(InputAction.DOWN), drawYCenter += (panelHeight))
+    {
+      // Only bother to draw it if it is onscreen.
+      if( (drawYCenter > -panelHeight/2) && ( drawYCenter < myHeight+(panelHeight/2) ) )
+      {
+        CommanderInfo coInfo = infos.get(coToDraw.getSelectionNormalized());
+        Integer key = new Integer(coToDraw.getSelectionNormalized());
+
+        // Get the relevant PlayerPanel.
+        if( !coPanels.containsKey(key) ) coPanels.put(key, new CommanderPanel(coInfo, playerColor));
+        CommanderPanel panel = coPanels.get(key);
+
+        // Update the PlayerPanel and render it to an image.
+        BufferedImage playerImage = panel.update(coInfo, playerColor);
+
+        int drawX = cursorSizePx * 2;
```

Comment:
Kind of a weird way to define spacing. Also it's the only usage of `cursorSizePx`.

Reply:
Hm. I might have been using it more places initially. At any rate, it guarantees enough space to draw the cursor, plus an appropriate buffer. Made sense to me at the time.